tests/Controller/ExportControllerTest.php

```diff
@@ -307,10 +310,189 @@ public function testExportWithValidArguments(
 
         $this->assertEquals(200, $result->getStatusCode());
         $xml = new SimpleXMLElement($result->getContent());
-        $this->assertSame(1, $xml->items->count());
+        $this->assertSame(1, (int)$xml->items[0]->attributes()['count']);
         $this->assertSame($productEntity->getId(), (string)$xml->items->item['id']);
     }
 
+    public function crossSellingCategoryProvider()
+    {
+        $categoryOne = '3f48c6c920e9972270b9730f97b395d2';
+        $categoryTwo = 'bd466201e971ead1265c2bb4ab72d5df';
+        $notInCrossSellingCategory = '0f26d5374acb10dddb2f3a027e92b85d';
+
+        return [
+            'No cross-sell categories configured' => [
+                'assignedCategory' => [
+                    [
+                        'id' => $notInCrossSellingCategory,
+                        'name' => 'NotInCrossSellingCategory'
+                    ]
+                ],
+                'crossSellingCategories' => [],
+                'expectedCount' => 1
+            ],
+            'Article does not exist in cross-sell category configured' => [
+                'assignedCategory' => [
+                    [
+                        'id' => $notInCrossSellingCategory,
+                        'name' => 'NotInCrossSellingCategory'
+                    ]
+                ],
+                'crossSellingCategories' => [$categoryOne],
+                'expectedCount' => 1
+            ],
+            'Article exists in one of the cross-sell categories configured' => [
+                'assignedCategory' => [
+                    [
+                        'id' => $categoryOne,
+                        'name' => 'Category 1'
+                    ]
+                ],
+                'crossSellingCategories' => [$categoryOne, $categoryTwo],
+                'expectedCount' => 0
+            ],
+            'Article exists in all of cross-sell categories configured' => [
+                'assignedCategory' => [
+                    [
+                        'id' => $categoryOne,
+                        'name' => 'Category 1',
+                    ],
+                    [
+                        'id' => $categoryTwo,
+                        'name' => 'Someothercategory',
+                    ]
+                ],
+                'crossSellingCategories' => [$categoryOne, $categoryTwo],
+                'expectedCount' => 0
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider crossSellingCategoryProvider
+     * @throws InconsistentCriteriaIdsException
+     * @throws UnknownShopkeyException
+     */
+    public function testExportWithCrossSellingCategories(
```

Comment:
Please find a way to create a better setup for this test. It is really hard to read and it is also very long. You can create some helper methods and move related methods in there.